lib/utils/find_best_index.ts

```diff
@@ -18,5 +23,11 @@ export default function (indexes: ModelIndexes, comparisonChart: ConditionStorag
 		.filter((index) => comparisonChart[index._hashKey]?.type === "EQ");
 
 	const index = validIndexes.find((index) => comparisonChart[index._rangeKey]) || validIndexes[0];
-	return index?.IndexName ?? null;
+
+	// If no index is found and hash key exists, use table index
+	if (!index && comparisonChart[tableHashKey]?.type === "EQ") {
+		return {"tableIndex": true};
+	}
+
+	return {"tableIndex": false, "indexName": index?.IndexName ?? null};
```

Comment:
One question. Without this change, is at least 1 of the tests you modified/added failing?

Reply:
Yep! Anything with `tableHashKey` should fail without this change.

https://github.com/andrewda/dynamoose/blob/42767a019c24902984e34387fa903e936b3ae2f1/test/unit/utils/find_best_index.js#L27-L40